cmd/builder/internal/builder/config.go

```diff
@@ -128,13 +129,18 @@ func (c *Config) ParseModules() error {
 	return nil
 }
 
-func parseModules(mods []Module) ([]Module, error) {
-	var parsedModules []Module
+func validateModules(mods []Module) error {
```

Comment:
As a new function, this might deserve its own tests.

Reply:
I disagree that we should have unittests for private methods. This is an implementation detail to avoid duplicate code.

https://softwareengineering.stackexchange.com/questions/100959/how-do-you-unit-test-private-methods

We do have unittest to check that `Validate` returns error if any of the Modules (extension, receiver, processor, exporter) are invalid.

Reply:
My 2 cents: IMO, if the private methods are fully testable via the public API calls then usually there is no need to test the private methods directly. Sometimes it is not possible or practical, and in such cases it is useful to cover the private methods by their own unit tests.

Reply:
I find it easier to exercise all the conditions by testing the functions in isolation, but given we have enough coverage here, it's fine to have it this way.